docs/generator_template.md

```diff
@@ -0,0 +1,569 @@
+
+Suppose you can only sleep when the AC in your bedroom is set to 22 °C, and you can't see when the temperature drops or rises above that. You can install a smart monitor in your bedroom that keeps track of the temperature and notifies you to adjust it to your optimum temperature if it fluctuates. You will create a template that alerts you when the temperature fluctuates from 22 °C.
+
+In this tutorial:
+
+- You'll use the [Eclipse Mosquito](https://test.mosquitto.org) **MQTT broker**, which you'll connect to subscribe and publish messages using an MQTT client.
+- You'll use [Python Paho-MQTT](https://pypi.org/project/paho-mqtt/) as the **MQTT client** in this project.
+- Lastly, you will create a React template that will use the MQTT broker to allow you to monitor your bedroom's temperature and notify you when the temperature drops or rises above 22 °C.
+- Create a reusable component for the output code's rise/dropped temperature functions.
+
+## Background context
+
+There is a list of [community maintained templates](https://www.asyncapi.com/docs/tools/generator/template#generator-templates-list), but what if you require customized output from the generator? In that case, you'll create a user-defined template that generates custom output from the generator.
+Before you create the template, you'll need to have an [AsyncAPI document](https://www.asyncapi.com/docs/tools/generator/asyncapi-document) that defines the properties you want to use in your template to test against. In this tutorial, you'll use the following template saved in the `test/fixtures/asyncapi.yml` file in your template project directory.
+
+``` yml
+
+asyncapi: 2.6.0
+
+info:
+  title: Temperature Service
+  version: 1.0.0
+  description: This service is in charge of processing all the events related to temperature.
+
+servers:
+  dev:
+    url: test.mosquitto.org
+    protocol: mqtt
+    bindings:
+      mqtt:
+        clientId: temperature-service
+
+channels:
+  temperature/changed:
+    description: Updates the bedroom temperature in the database when the temperatures drops or goes up.
+    publish:
+      operationId: temperatureChange
+      message:
+        description: Message that is being sent when the temperature in the bedroom changes.
+        payload:
+          type: object
+          additionalProperties: false
+          properties:
+            temperatureId:
+              type: string
+components:
+  schemas:
+    temperatureId:
+      type: object
+      additionalProperties: false
+      properties:
+        temperatureId:
+          type: string
+```
+
+Note:
+
+- To generate code, use the [AsyncAPI CLI](https://www.asyncapi.com/tools/cli). If you don't have the CLI installed, follow [this guide](https://www.asyncapi.com/docs/tools/generator/installation-guide#asyncapi-cli).
```

Comment:
since this document is rendered in website, and people should read it in the website, you can do relative website links

```suggestion
- To generate code, use the [AsyncAPI CLI](https://www.asyncapi.com/tools/cli). If you don't have the CLI installed, follow [CLI installation guide](/docs/tools/cli/installation).
```

- better point to CLI docs
- better links if they are descriptive for people with accessibility problems, that use screen readers